Incorrect return type for flush() function

**Describe the bug**
The type declaration of the onDebouncedCallback `flush()` function claims that the return type is `void`. However, that's incorrect: the actual return type is the same as the callback. (Or `undefined` if the callback has never been called yet.) This causes Typescript errors when using `flush()`.

**To Reproduce**
```js
const foo = useDebouncedCallback(() => 'abc', 500);
foo();
const result: string | undefined = foo.flush();
```

results in:

```
foobar.tsx:31:11 - error TS2322: Type 'void' is not assignable to type 'string | undefined'.

31     const result: string | undefined = foo.flush();
```

**Expected behavior**
Typescript type declaration should match the function's actual type.

**use-debounce version:** 10.0.0

--- a/src/useDebouncedCallback.ts
+++ b/src/useDebouncedCallback.ts
@@ -22,7 +22,7 @@ export interface Options extends CallOptions {
   debounceOnServer?: boolean;
 }
 
-export interface ControlFunctions {
+export interface ControlFunctions<ReturnT> {
   /**
    * Cancel pending function invocations
    */
@@ -30,7 +30,7 @@ export interface ControlFunctions {
   /**
    * Immediately invoke pending function invocations
    */
-  flush: () => void;
+  flush: () => ReturnT | undefined;
   /**
    * Returns `true` if there are any pending function invocations
    */
@@ -42,7 +42,7 @@ export interface ControlFunctions {
  * Note, that if there are no previous invocations you will get undefined. You should check it in your code properly.
  */
 export interface DebouncedState<T extends (...args: any) => ReturnType<T>>
-  extends ControlFunctions {
+  extends ControlFunctions<ReturnType<T>> {
   (...args: Parameters<T>): ReturnType<T> | undefined;
 }
 

--- a/test/useDebouncedCallback.test.tsx
+++ b/test/useDebouncedCallback.test.tsx
@@ -500,7 +500,7 @@ describe('useDebouncedCallback', () => {
 
       debounced();
       useEffect(() => {
-        return debounced.flush;
+        return () => { debounced.flush(); };
       }, []);
       return <span role="test">{text}</span>;
     }

